sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -330,6 +339,15 @@ case class PartitioningCollection(partitionings: Seq[Partitioning])
   override def satisfies0(required: Distribution): Boolean =
     partitionings.exists(_.satisfies(required))
 
+  override def createShuffleSpec(
+      defaultNumPartitions: Int,
+      distribution: ClusteredDistribution): ShuffleSpec = {
+    require(satisfies(distribution), "createShuffleSpec should only be called after satisfies " +
```

Comment:
shall we let the caller side do this check? it's a bit weird that only `PartitioningCollection` have this check.

Reply:
OK, let me move this check to `EnsureRequirements`.